packages/dd-trace/src/appsec/iast/taint-tracking/operations-taint-object.js

```diff
@@ -40,6 +40,55 @@ function taintObject (iastContext, object, type) {
   return result
 }
 
+function taintQueryWithCache (iastContext, query, type) {
+  const transactionId = iastContext?.[IAST_TRANSACTION_ID]
+  if (!transactionId || !query) return
+
+  if (!iastContext.queryCache) {
+    iastContext.queryCache = {}
+  }
+
+  return traverseAndTaint(query, iastContext.queryCache, '', type, transactionId)
+}
+
+function traverseAndTaint (value, cache, path, type, transactionId) {
+  if (value == null) return value
+
+  if (typeof value === 'string') {
+    // If we already have a tainted version of this exact string in the cache
+    if (value === cache) {
+      return cache
+    }
+
+    return TaintedUtils.newTaintedString(transactionId, value, path, type)
+  }
+
+  if (typeof value === 'object') {
+    const isArray = Array.isArray(value)
```

Comment:
I don't like the name `isArray` which is like the function name. Maybe something like `valueIsArray` or `isValueArray`?